File: generators/c_stdlib_rand/generator.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>


const int BITSTRING_SIZE = 16000000;


int main() 
{
    srand(time(NULL));
    FILE *file = fopen("randomness.txt", "wb");

    for (int i = 0; i < BITSTRING_SIZE / 4; i++) {
        int r = rand();
        fwrite(&r, sizeof(r), 1, file);
    }

    fclose(file);
    return 0;
}

File: generators/matlab_randi/generator.m
function generator()
    BITSTRING_SIZE = 16000000;
    outfile = fopen("randomness.txt", "w");

    for i = 1:BITSTRING_SIZE/8
        r = randi(intmax("uint32"));
        bytes = typecast(r, 'uint8');
        fwrite(outfile, bytes, 'uint8');
    end

    flose(outfile);
end